term-utils/scriptreplay.c

```diff
@@ -308,9 +320,64 @@ main(int argc, char *argv[])
 		replay_set_delay_max(setup, &maxdelay);
 	replay_set_delay_min(setup, &mindelay);
 
-	isterm = setterm(&saved);
+	isterm = setterm(&saved, &saved_flag);
 
 	do {
+		switch (fgetwc(stdin)) {
```

Comment:
There is include/fgetwc_or_err.h with fgetwc_or_err(). I guess it can increase the reliability of the code.

Reply:
IMO we should not use fgetwc_or_err as we expect fgetwc to have many instances of WEOF with errno !=0 when no user input is added to the stream.
```c
// section from fgetwc_or_err
if (ret == WEOF && errno != 0)
		err(EXIT_FAILURE, _("fgetwc() failed"));
```
We run at non-blocking mode and this results in this error showing up every time no input is found `getwc() failed: Resource temporarily unavailable`, which is very common and repeats often during a replay.

Reply:
Good point, I forgot about the non-blocking mode ...

Reply:
BTW, why do we need widechar (fgetwc() and wchat_t)  here? I guess it will work with arbitrary function like fgetc().